api/pkg/transformer/symbol/time_test.go

```diff
@@ -0,0 +1,658 @@
+package symbol
+
+import (
+	"testing"
+	"time"
+
+	"github.com/gojek/merlin/pkg/transformer/jsonpath"
+	"github.com/gojek/merlin/pkg/transformer/types/series"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestSymbolRegistry_ParseTimestamp(t *testing.T) {
+	testCases := []struct {
+		desc          string
+		timestamp     interface{}
+		requestJSON   []byte
+		responseJSON  []byte
+		expectedValue interface{}
+		err           error
+	}{
+		{
+			desc:          "Using literal value (int64) for timestamp",
+			timestamp:     1619541221,
+			expectedValue: time.Date(2021, 4, 27, 16, 33, 41, 0, time.UTC),
+		},
+		{
+			desc:          "Using literal value (string) for timestamp",
+			timestamp:     "1619541221",
+			expectedValue: time.Date(2021, 4, 27, 16, 33, 41, 0, time.UTC),
+		},
+		{
+			desc:      "Using single value request jsonPath for timestamp",
+			timestamp: "$.timestamp",
+			requestJSON: []byte(`{
+				"timestamp": 1619541221
+			}`),
+			expectedValue: time.Date(2021, 4, 27, 16, 33, 41, 0, time.UTC),
+		},
+		{
+			desc:      "Using single value response jsonPath for timestamp",
+			timestamp: "$.model_response.timestamp",
+			responseJSON: []byte(`{
+				"timestamp": 1619541221
+			}`),
+			expectedValue: time.Date(2021, 4, 27, 16, 33, 41, 0, time.UTC),
+		},
+		{
+			desc:      "Using array from request jsonpath for timestamp",
+			timestamp: "$.timestamps[*]",
+			requestJSON: []byte(`{
+				"timestamps": [1619541221, 1619498021]
+			}`),
+			expectedValue: []interface{}{
+				time.Date(2021, 4, 27, 16, 33, 41, 0, time.UTC),
+				time.Date(2021, 4, 27, 4, 33, 41, 0, time.UTC),
+			},
+		},
+		{
+			desc:      "Using array from request jsonpath for timestamp - different type",
+			timestamp: "$.timestamps[*]",
+			requestJSON: []byte(`{
+				"timestamps": [1619541221, "1619498021"]
+			}`),
+			expectedValue: []interface{}{
+				time.Date(2021, 4, 27, 16, 33, 41, 0, time.UTC),
+				time.Date(2021, 4, 27, 4, 33, 41, 0, time.UTC),
+			},
+		},
+		{
+			desc:      "Using series",
+			timestamp: series.New([]interface{}{1619541221, 1619498021}, series.Int, "timestamp"),
+			expectedValue: []interface{}{
+				time.Date(2021, 4, 27, 16, 33, 41, 0, time.UTC),
+				time.Date(2021, 4, 27, 4, 33, 41, 0, time.UTC),
+			},
+		},
+	}
+	for _, tC := range testCases {
+		t.Run(tC.desc, func(t *testing.T) {
+			sr := NewRegistryWithCompiledJSONPath(jsonpath.NewStorage())
+
+			requestJSONObj, responseJSONObj := createTestJSONObjects(tC.requestJSON, tC.responseJSON)
+			if requestJSONObj != nil {
+				sr.SetRawRequestJSON(requestJSONObj)
+			}
+			if responseJSONObj != nil {
+				sr.SetModelResponseJSON(responseJSONObj)
+			}
+
+			got := sr.ParseTimestamp(tC.timestamp)
+			assert.Equal(t, tC.expectedValue, got)
+		})
+	}
+}
+
+func TestSymbolRegistry_IsWeekend(t *testing.T) {
```

Comment:
Can you add test where input is a `Series`?

Reply:
Tio already added that: https://github.com/gojek/merlin/blob/std-trf-built-in-function/api/pkg/transformer/symbol/time_test.go#L117-L172